### PR TITLE
Add linux install script for OPW

### PR DESCRIPTION
**Pull request by spencergilbert**
Re-opening [original PR](https://github.com/DataDog/agent-linux-install-script/pull/63#issue-1805511510) after it stopped noticing my commits...

### CODE REVIEW

File: install_script_op_worker.sh [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+    OS="Debian"
+elif [ -f /etc/redhat-release ] || [ "$DISTRIBUTION" == "RedHat" ] || [ "$DISTRIBUTION" == "CentOS" ] || [ "$DISTRIBUTION" == "Amazon" ]; then
+    OS="RedHat"
+# Some newer distros like Amazon may not have a redhat-release file
+elif [ -f /etc/system-release ] || [ "$DISTRIBUTION" == "Amazon" ]; then
+    OS="RedHat"
+fi
+
+# Root user detection
+if [ "$(echo "$UID")" = "0" ]; then
```
Comment (chouetz) [inline install_script_op_worker.sh:303]:
```suggestion
if [ "$(echo "$UID")" == "0" ]; then
```
according to google shell [guide](https://google.github.io/styleguide/shellguide.html#testing-strings)

File: install_script_op_worker.sh [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+    exit 1;
+  fi
+fi
+
+# OS/Distro Detection
+# Try lsb_release, fallback with /etc/issue then uname command
+KNOWN_DISTRIBUTION="(Debian|Ubuntu|RedHat|CentOS|Amazon)"
+DISTRIBUTION=$(lsb_release -d 2>/dev/null | grep -Eo $KNOWN_DISTRIBUTION  || grep -Eo $KNOWN_DISTRIBUTION /etc/issue 2>/dev/null || grep -m1 -Eo $KNOWN_DISTRIBUTION /etc/os-release 2>/dev/null || uname -s)
+
+if [ "$DISTRIBUTION" = "Darwin" ]; then
```
Comment (chouetz) [inline install_script_op_worker.sh:287]:
```suggestion
if [ "$DISTRIBUTION" == "Darwin" ]; then
```

File: install_script_op_worker.sh [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+
+# Root user detection
+if [ "$(echo "$UID")" = "0" ]; then
+    sudo_cmd=''
+else
+    sudo_cmd='sudo'
+fi
+
+# Install the necessary package sources
+if [ "$OS" = "RedHat" ]; then
```
Comment (chouetz) [inline install_script_op_worker.sh:310]:
```suggestion
if [ "$OS" == "RedHat" ]; then
```

File: install_script_op_worker.sh [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+    fi
+
+    declare -a packages
+    packages=("$worker_flavor")
+    
+    echo -e "  \033[33mInstalling package(s): ${packages[*]}\n\033[0m"
+
+    $sudo_cmd yum -y --disablerepo='*' --enablerepo='observability-pipelines-worker' install $dnf_flag "${packages[@]}" || $sudo_cmd yum -y install $dnf_flag "${packages[@]}"
+
+elif [ "$OS" = "Debian" ]; then
```
Comment (chouetz) [inline install_script_op_worker.sh:366]:
```suggestion
elif [ "$OS" == "Debian" ]; then
```

**Review by chouetz — COMMENTED** [review]
I guess this will need a QA campaign to validate

File: install_script_op_worker.sh (hunk 4 above)
Comment (spencergilbert) [reply install_script_op_worker.sh]: Addressed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/5b76b4d796bcc42eb3364ed7cd8fc9e9a33fe997/https://github.com/DataDog/agent-linux-install-script/pull/66/commits/707be0c4f2c8d66fe3ad78290f755dd6102a5725

File: install_script_op_worker.sh (hunk 1 above)
Comment (spencergilbert) [reply install_script_op_worker.sh]: Addressed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/5b76b4d796bcc42eb3364ed7cd8fc9e9a33fe997 / https://github.com/DataDog/agent-linux-install-script/pull/66/commits/707be0c4f2c8d66fe3ad78290f755dd6102a5725

File: test/dockertest.sh [hunk 5]
```diff
@@ -46,6 +46,9 @@ while [[ $# -gt 0 ]]; do
     --yum-version-path)
       TESTING_YUM_VERSION_PATH="$2"
       ;;
+    --observability-pipelines-worker)
+      DD_OP="$2"
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply test/dockertest.sh]: Addressed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/5b76b4d796bcc42eb3364ed7cd8fc9e9a33fe997 / https://github.com/DataDog/agent-linux-install-script/pull/66/commits/707be0c4f2c8d66fe3ad78290f755dd6102a5725

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -172,3 +189,8 @@ deploy_7:
   extends: .deploy
   script:
     - $S3_CP_CMD ./install_script_agent7.sh s3://dd-agent/scripts/install_script_agent7.sh --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
+
+deploy_opw:
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply .gitlab-ci.yml]: Addressed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/5b76b4d796bcc42eb3364ed7cd8fc9e9a33fe997 / https://github.com/DataDog/agent-linux-install-script/pull/66/commits/707be0c4f2c8d66fe3ad78290f755dd6102a5725

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -148,6 +149,22 @@ test-apm-injection:
...
     - ./test/dockertest.sh --image "registry.ddbuild.io/${IMAGE}" --script install_script_agent7.sh --injection "$INJECTION" --no-agent "$NO_AGENT" --apm-libraries "$APM_LIBRARIES"
 
+test-observability-pipelines-worker:
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/docker_x64:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["runner:docker"]
+  stage: test
+  dependencies: ["generate-scripts"]
+  parallel:
+    matrix:
+      - IMAGE: images/mirror/amazonlinux:2
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply .gitlab-ci.yml]: Addressed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/5b76b4d796bcc42eb3364ed7cd8fc9e9a33fe997 / https://github.com/DataDog/agent-linux-install-script/pull/66/commits/707be0c4f2c8d66fe3ad78290f755dd6102a5725

[the remark this answers is not in the document]
Comment (spencergilbert) [reply .gitlab-ci.yml]: Addressed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/5b76b4d796bcc42eb3364ed7cd8fc9e9a33fe997 / https://github.com/DataDog/agent-linux-install-script/pull/66/commits/707be0c4f2c8d66fe3ad78290f755dd6102a5725

**Review by bkabrda — CHANGES_REQUESTED** [review]
:wave: thanks a lot for this PR. It looks really solid, I just left couple minor comments inline to get addressed and then we can merge.

File: install_script_op_worker.sh [hunk 8]
```diff
@@ -0,0 +1,572 @@
...
+    # repo_gpgcheck on RHEL/CentOS 8.1
+    if [ -z "$rpm_repo_gpgcheck" ]; then
+        if grep -q "8\.1\(\b\|\.\)" /etc/redhat-release 2>/dev/null; then
+            rpm_repo_gpgcheck=0
+        else
+            rpm_repo_gpgcheck=1
+        fi
+    fi
+
+    gpgkeys="https://${keys_url}/DATADOG_RPM_KEY.public\n       https://${keys_url}/DATADOG_RPM_KEY_E09422B3.public"
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply install_script_op_worker.sh:330]: Agreed - I remember this one puzzling me, as I think [the same thing](https://github.com/DataDog/agent-linux-install-script/blob/1.18.0/install_script.sh.template#L717-L724) happens with the Agent I think?

File: install_script_op_worker.sh
[the remark this answers is not in the document]
Comment (spencergilbert) [reply install_script_op_worker.sh:1]: Renamed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/4907e5a3ed50cc9c46b9283b04838a8fa954a4a3

File: install_script_op_worker.sh [hunk 9]
```diff
@@ -0,0 +1,572 @@
...
+        elif [ $apt_exit_code -ne 0 ]; then
+            cat $DD_APT_INSTALL_ERROR_MSG
+            exit $apt_exit_code
+        else
+            break
+        fi
+    done
+
+    printf "\033[34m\n* Installing APT package sources for Datadog\n\033[0m\n"
+    $sudo_cmd sh -c "echo 'deb [signed-by=${apt_usr_share_keyring}] https://${apt_url}/ ${apt_repo_version}' > /etc/apt/sources.list.d/datadog.list"
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply install_script_op_worker.sh]: Fixed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/4069004ab1b44e6f06e96fc1dcc0fec0ab8a5de3

File: install_script_op_worker.sh (hunk 8 above)
[the remark this answers is not in the document]
Comment (spencergilbert) [reply install_script_op_worker.sh]: Fixed in https://github.com/DataDog/agent-linux-install-script/pull/66/commits/4069004ab1b44e6f06e96fc1dcc0fec0ab8a5de3

File: .gitlab-ci.yml [hunk 10]
```diff
@@ -162,5 +180,6 @@ test-apm-injection:
       - SCRIPT: install_script.sh
       - SCRIPT: install_script_agent6.sh
       - SCRIPT: install_script_agent7.sh
+      - SCRIPT: install_script_opw1.sh
```
[the remark this answers is not in the document]
Comment (spencergilbert) [reply .gitlab-ci.yml:183]: Ah, yep - might have been a poor conflict resolution.

[the remark this answers is not in the document]
Comment (spencergilbert) [reply .gitlab-ci.yml]: Fixed https://github.com/DataDog/agent-linux-install-script/pull/66/commits/7a870d92bb20619ac539f520d35b5f02ac22dac6

**Review by bkabrda — APPROVED** [review]

**Pull request merged**